khelo/lib/ui/flow/tournament/detail/tournament_detail_view_model.dart

```diff
@@ -43,19 +49,31 @@ class TournamentDetailStateViewNotifier
 
     state = state.copyWith(loading: true);
 
-    _tournamentSubscription = _tournamentService
-        .streamTournamentById(_tournamentId!)
-        .listen((tournament) {
-      state = state.copyWith(
-        tournament: tournament,
-        loading: false,
-      );
-      onMatchFilter(null);
-      onStatFilter(state.selectedFilterTag);
+    final tournamentData = combineLatest3(
+        _tournamentService.streamTournamentById(_tournamentId!),
+        _tournamentService.streamTeamStats(_tournamentId!),
+        _tournamentService.streamPlayerKeyStats(_tournamentId!));
+
+    tournamentData.listen((data) async {
+      final tournament = data.$1;
+      final teamStats = data.$2;
+      final keyStats = data.$3.getTopKeyStats();
+
+      _matchService.streamMatchesByIds(tournament.match_ids).listen((matches) {
+        state = state.copyWith(
+          tournament: tournament,
+          teamStats: teamStats,
+          keyStats: keyStats,
+          matches: matches,
+          loading: false,
+        );
+        onMatchFilter(null);
+        onStatFilter(state.selectedFilterTag);
+      });
```

Comment:
we're setting listener inside listener without any condition to prevent listening same stream multiple times, can you fix it, can we simplify it somehow?